### ISSUE TITLE
Unexpected results on training the model on single class

### ISSUE THREAD

**Issue by Siddhijain16** [facebookresearch/detectron2#2408]
I trained the model using COCO-Detection/faster_rcnn_X_101_32x8d_FPN_3x.yaml, on the single class with my custom dataset in which I took 400 imgs for training 100 for testing & 100 for validation.
I follow the flow of https://colab.research.google.com/drive/1-TNOcPm3Jr3fOJG8rnGT9gh60mHUsvaW#scrollTo=wXisIbT1Zqou
(Roboflow-Custom-Detectron2.ipynb)
I also changed the threshold ,learning rate according to my data but I got this type of result after training is this because of model error or code error ? Should I need to do some transformation on data before passing to the model ??
 
#from .detectron2.tools.train_net import Trainer
#from detectron2.engine import DefaultTrainer
from detectron2.config import get_cfg
#from detectron2.evaluation.coco_evaluation import COCOEvaluator
import os
from detectron2.checkpoint import DetectionCheckpointer, PeriodicCheckpointer
from detectron2.modeling import build_model

cfg = get_cfg()
cfg.merge_from_file(model_zoo.get_config_file("COCO-Detection/faster_rcnn_X_101_32x8d_FPN_3x.yaml"))
cfg.DATASETS.TRAIN = ("my_dataset_train",)
cfg.DATASETS.TEST = ("my_dataset_val",)

cfg.DATALOADER.NUM_WORKERS = 4
cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url("COCO-Detection/faster_rcnn_X_101_32x8d_FPN_3x.yaml")  # Let training initialize from model zoo
cfg.SOLVER.IMS_PER_BATCH = 4
cfg.SOLVER.BASE_LR = 0.0000001

cfg.SOLVER.WARMUP_ITERS = 1000
cfg.SOLVER.MAX_ITER = 1500 #adjust up if val mAP is still rising, adjust down if overfit
cfg.SOLVER.STEPS = (1000, 1500)
cfg.SOLVER.GAMMA = 0.05
#cfg.TEST.DETECTIONS_PER_IMAGE = 5

#model = build_model(cfg)
#checkpointer = DetectionCheckpointer(model).load(cfg.MODEL.WEIGHTS)
#checkpointer = DetectionCheckpointer(model, save_dir="/content/drive/MyDrive/detectron2/")
#checkpointer = PeriodicCheckpointer(checkpointer, period=100,max_iter = 500)
#checkpointer.save("model_new_robocheck2")

cfg.MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE = 64
cfg.MODEL.ROI_HEADS.NUM_CLASSES = 1 

cfg.TEST.EVAL_PERIOD = 500

os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)
trainer = CocoTrainer(cfg)
trainer.resume_or_load(resume=False)
trainer.train()

![image](https://user-images.githubusercontent.com/46879959/102875990-96134700-446a-11eb-9ad4-f7c44f50d25a.png)

![image](https://user-images.githubusercontent.com/46879959/102876088-c0650480-446a-11eb-9f05-ff64d9160fb1.png)

please help me

**Comment by ppwwyyxx** [comment]
As the [issue template](https://github.com/facebookresearch/detectron2/blob/master/.github/ISSUE_TEMPLATE/unexpected-problems-bugs.md) mentions:

> If you expect the model to converge / work better, note that we do not give suggestions on how to train a new model. Only in one of the two conditions we will help with it: (1) You're unable to reproduce the results in detectron2 model zoo. (2) It indicates a detectron2 bug.

We provide configs & models with standard academic settings and expect users to have the knowledge to choose or design appropriate models & parameters for their own tasks.
